include/uv-errno.h

```diff
@@ -22,7 +22,16 @@
 #ifndef UV_ERRNO_H_
 #define UV_ERRNO_H_
 
-#include <errno.h>
+#ifndef _WIN32
+# include <errno.h>
+# if EDOM > 0
+#  define UV_ERR(x) (-(x))
+# else
+#  define UV_ERR(x) (x)
+# endif
+#else
+# define UV_ERR(x)  (x)
+#endif
```

Comment:
Can you undef the macro again at the end of include/uv.h?  Maybe move its definition to that file if you think that's cleaner.

For the files in src/unix, you'll want a duplicate definition in src/unix/internal.h.  Duplication is bad but not leaking symbols into the libuv user's namespace outweighs that, IMO.

Reply:
Do I need to undefine `UV_ERR()` in `src/unix/internal.h`? Or should it be alright without undefining since this is an internal header?

Reply:
Sorry, bad question - `UV_ERR()` is needed in the files.